cg/meta/transfer/external_data.py

```diff
@@ -0,0 +1,202 @@
+"""Module for deliver and rsync customer inbox on hasta to customer inbox on caesar"""
+import datetime as dt
+import logging
+from pathlib import Path
+from typing import List, Tuple, Optional
+
+from housekeeper.store.models import Bundle
+
+from cg.apps.cgstats.db.models import Version
+from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.apps.slurm.slurm_api import SlurmAPI
+from cg.meta.meta import MetaAPI
+from cg.meta.rsync.sbatch import RSYNC_COMMAND, ERROR_RSYNC_FUNCTION
+from cg.models.cg_config import CGConfig
+from cg.models.slurm.sbatch import Sbatch
+from cg.store import models
+from cg.meta.transfer.md5sum import check_md5sum, extract_md5sum
+from cg.constants import HK_FASTQ_TAGS
+
+LOG = logging.getLogger(__name__)
+
+
+class ExternalDataAPI(MetaAPI):
+    def __init__(self, config: CGConfig):
+        super().__init__(config)
+        self.hasta_path: str = config.external.hasta
+        self.caesar_path: str = config.external.caesar
+        self.base_path: str = config.data_delivery.base_path
+        self.account: str = config.data_delivery.account
+        self.mail_user: str = config.data_delivery.mail_user
+        self.housekeeper_api: HousekeeperAPI = config.housekeeper_api
+
+    def create_log_dir(self, dry_run: bool, ticket_id: int) -> Path:
+        """Creates a directory for log file to be stored"""
+        timestamp: dt.datetime = dt.datetime.now()
+        timestamp_str: str = timestamp.strftime("%y%m%d_%H_%M_%S_%f")
+        folder_name: Path = Path("_".join([str(ticket_id), timestamp_str]))
+        log_dir: Path = Path(self.base_path) / folder_name
+        LOG.info("Creating folder: %s", log_dir)
+        if dry_run:
+            LOG.info("Would have created path %s, but this is a dry run", log_dir)
+            return log_dir
+        log_dir.mkdir(parents=True, exist_ok=False)
+        return log_dir
+
+    def create_source_path(
+        self, cust_sample_id: str, customer: str, raw_path: str, ticket_id: int
+    ) -> Path:
+        """Returns the path to where the sample files are fetched from"""
+        return Path(raw_path % customer).joinpath(str(ticket_id), cust_sample_id)
+
+    def create_destination_path(self, customer: str, lims_sample_id: str, raw_path: str) -> Path:
+        """Returns the path to where the files are to be transferred"""
+        return Path(raw_path % customer).joinpath(lims_sample_id)
```

Comment:
those are not creating a path! So shouldnt be names create

Reply:
and if you use string formatting in path, it could be nice to still move out the path structure as a property, then apply cormatting with args later when its used